packages/gatsby-source-mongodb/src/gatsby-node.js

```diff
@@ -71,14 +71,14 @@ function createNodes(
         reject(err)
       }
 
-      documents.forEach(item => {
-        var id = item._id.toString()
+      documents.map(d => JSON.parse(JSON.stringify(d))).forEach(item => {
+        const id = item._id
         delete item._id
 
-        var node = {
+        const node = {
           // Data for the node.
           ...item,
-          id: createNodeId(`${id}`),
+          id: id,
```

Comment:
we want to use `createNodeId` - what was probably missing is we didn't use that for in other parts when we reference other nodes

Reply:
the issue is that when the reference is on the mongoDB side, the reference uses `_id`. createNodeId is not idempotent, so there's no way to know what it will be beforehand. A workaround (that I haven't tested yet) would be to name the field in the database
```diff
- field___NODE
+ field___NODE___mongo_id
```

Reply:
so, you have actual `field___NODE` field in your db? so that `___NODE` isn't added by plugin? need to understand what this is actually fixing

Reply:
Correct, it's directly in the DB. I have a foreign key in my database which points to a shared resource. 
As it stands, `findLinkedNodes` will try to connect `field___NODE` with `id`, but field___NODE will never match `id` since it's generated by `createNodeId`

Reply:
Of course, ultimately, if you feel this PR won't be beneficial to the community, I'm okay with making a specialized source plugin for my project 😄

Reply:
I would be very interested too see if this:
```diff
- field___NODE
+ field___NODE___mongo_id
```
would make it work - we have some code to handle this notation, but noone really use that and at this point it might actually not work (AFAIK we don't have tests for that). 

We added `createNodeId` stuff because there were issues with `id` namespace collisions (multiple plugins adding same id, which cause weird bugs), so I don't think we want to get rid of that. Potentially we could do some post processing on data and apply `createNodeId` to raw data if field ends with `___NODE` in plugin, but this is real edge case - I don't think many people would use `___NODE` notation directly in mongodb. Maybe this could be added alongisde `mapping` - ability to define node links in `gatsby-config` options? (just thinking out loud here)